src/lib/components/Swap/Toolbar/index.tsx

```diff
@@ -42,11 +45,10 @@ export default function Toolbar({ disabled }: { disabled?: boolean }) {
       return <Caption.InsufficientBalance currency={trade.inputAmount.currency} />
     }
 
-    if (inputCurrency && outputCurrency) {
+    if (inputCurrency && outputCurrency && (swapInputAmount || swapOutputAmount)) {
```

Comment:
What you actually want here is whether there is a populated amount at all. Could you make a new small hook in `lib/hooks/swap` to do that? It only needs to look at `useAtomValue(amountAtom)`, so it should have less churn.

Reply:
Yep makes sense